cirq-core/cirq/experiments/measure_in_random_bases.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2024 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections.abc import Mapping, Sequence
+from typing import Any, Literal
+
+import cirq
+import numpy as np
+import numpy.typing as npt
+
+
+class RandomizedMeasurements:
+    def __init__(
+        self,
+        num_qubits: int,
+        num_unitaries: int,
+        subsystem: Sequence[str | int] | None = None,
+        qubit_mapping: Mapping[int, str | int] | None = None,
+        rng: np.random.Generator | None = None,
```

Comment:
I know it's a common code pattern in Cirq but it's a harmful practice. please pass `rng` to `unitaries_to_moment` instead